hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackHelper.java

```diff
@@ -145,6 +165,17 @@ public ListingBasedRollbackHelper(HoodieTableMetaClient metaClient, HoodieWriteC
       }
       return false;
     };
-    return fs.listStatus(FSUtils.getPartitionPath(config.getBasePath(), partitionPath), filter);
+    return metaClient.getFs().listStatus(FSUtils.getPartitionPath(config.getBasePath(), partitionPath), filter);
+  }
+
+  private FileStatus[] getFilesFromCommitMetadata(HoodieCommitMetadata commitMetadata, String partitionPath) throws IOException {
+    HashSet<String> fullPaths = commitMetadata.getFullPathsByPartitionPath(metaClient.getBasePath(), partitionPath);
+    List<String> commitPathNames = fullPaths.stream().filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
can filter null from inside `getFullPathsByPartitionPath` ? so caller does not need to worry about null items.